scripts/lua2dox.lua

```diff
@@ -54,6 +54,7 @@ The effect is that you will get the function documented, but not with the parame
 ]]
 
 local TYPES = { 'integer', 'number', 'string', 'table', 'list', 'boolean', 'function' }
+local TYPE_PAT = '[a-zA-Z][a-zA-Z_.]*'
```

Comment:
This is already a great improvement, but it seems that more complex types like `any[]|nil` are not captured.

```suggestion
local TYPE_PAT = '[a-zA-Z][a-zA-Z_.%[%]]*'
```

With the above we can deal with `foo[]|nil` types. Function types that contain `?`, `,` `(`, `)`, etc. might be quite complicated to process with regex (we'll need treesitter eventually....), so this might be addressed as a separate scope.

Reply:
We can also capture the `optional?` types:

```suggestion
local TYPE_PAT = '[a-zA-Z][a-zA-Z_.%[%]|?]*'
```
There are only a few such occurrences, `(string?)`, `(Version ?|nil)`, etc.

Reply:
I don't think treesitter would ever be the right tool for this. Instead I've implemented an lpeg grammar to capture many more types.

Reply:
FWIW, (unlocking) improvements in our tooling is a valid argument for standardizing type annotations.